oss-licenses-plugin/src/main/groovy/com/google/android/gms/oss/licenses/plugin/LicensesTask.groovy

```diff
@@ -216,16 +217,19 @@ class LicensesTask extends DefaultTask {
             return
         }
 
-        String licenseKey = "${group}:${name}"
+        String libraryName = rootNode.name
```

Comment:
I did some further testing and unfortunately the name field is not sufficiently unique to provide a lookup. Maybe just switch to add the test fixes and deduplication here to #190. I'm OOO next week, but can take a look when I return.

In my testing with github.com/googlesamples/task-interop using this PR, two sets of SDKs collided on their name field which hides their licenses in an unpredictable way. You can see this by looking at app/build/generated/third_party_licenses/res/raw/third_party_licenses_metadata and comparing the output before and after this PR.

Example 1: https://maven.google.com/web/index.html?q=androidx.ac#androidx.activity:activity:1.3.1 and https://maven.google.com/web/index.html?q=savedstat#androidx.savedstate:savedstate:1.1.0 both have a POM Name of "Activity". This clobbers the licenses for one of them as they share the same key.

Example 2: https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-core-metadata/1.4.2/kotlinx-coroutines-core-metadata-1.4.2.pom and https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-android/1.5.1/kotlinx-coroutines-android-1.5.1.pom both have a POM Name of "kotlinx-coroutines-android". Again one will always clobber the other. 

There are probably a lot of examples of this sort of collision as the name isn't guaranteed to be unique while the artifact's maven coordinates are.

Reply:
That is interesting what you find out. I will review this and fix my new bugs.

Reply:
Ah crap. Resolving conversation does not submit the comment.
I did not intend to dedublicate the licenses, so I rolled back my changes a bit that the library's group and artifact id is used again. I merged #190 into this so everything should be fine now